control_client/src/main.c

```diff
@@ -121,35 +85,16 @@ int main(int argc, char **argv) {
                 ;
         }
 
-        for (unsigned int i = 0; i < sizeof(commands) / sizeof(commands[0]); i++) {
-            
-            if (commands[i].key == key) {
-                header_p hdr = {.type = TYPE_CNTRL, .subtype = commands[i].subtype}; // create header
-
-                // check what type of command it is
-                switch (commands[i].subtype) {
-                case CNTRL_ACT_REQ:
-                    switch_t *actuator = commands[i].priv; // get the actuator data
-                    actuator->state = !actuator->state;    // flip the state
-                    act_req_p act_req = {.id = actuator->act_id, .state = actuator->state}; // Create message
-
-                    pad_send(&pad, &hdr, sizeof(hdr));
-                    pad_send(&pad, &act_req, sizeof(act_req));
-                    break;
-                case CNTRL_ARM_REQ:
-                    uint8_t *level = commands[i].priv; // get arming level data
-                    arm_req_p arm_req = {.level = *level}; // create message
```

Comment:
Lol my bad, this is the line that causes the segfault. I forgot that you are storing `priv` as `void*` This means that when `*level` gets de-referenced, it's sometimes de-referencing a NULL pointer. Replace just `*level` with `(uint8_t)level` and ignore the casting warnings. My bad for the miscommunication lol, no need to revert the entire commit!